Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1893,14 +1893,31 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
       return .skipChildren
     }
 
-    if let trailingComponents = node.unexpectedAfterComponents,
-      let components = node.components
-    {
-      addDiagnostic(
-        trailingComponents,
-        TrailingVersionAreIgnored(major: node.major, components: components),
-        handledNodes: [trailingComponents.id]
-      )
+    if let unexpectedAfterComponents = node.unexpectedAfterComponents {
+      if let components = node.components,
+        unexpectedAfterComponents.allSatisfy({ $0.is(VersionComponentSyntax.self) })
+      {
+        addDiagnostic(
+          unexpectedAfterComponents,
+          TrailingVersionAreIgnored(major: node.major, components: components),
+          handledNodes: [unexpectedAfterComponents.id]
+        )
+      } else {
+        var handledNodes: [SyntaxIdentifier] = [unexpectedAfterComponents.id]
+        if node.major.hasError {
+          handledNodes.append(node.major.id)
+        }
+        if let components = node.components,
+          components.hasError
+        {
+          handledNodes.append(components.id)
+        }
```

Comment:
I think you should be able to just unconditionally add `node.major.id` and `components.id` to `handledNodes`

Reply:
That's right, it's behaving just like adding always 😅